docs-gb/examples/pipeline-cyclic.md

```diff
@@ -0,0 +1,358 @@
+        self.params = settings.parameters
+        self.extra = self.params.extra if self.params is not None else None
+        self.delay = self.extra.get("delay", 0)
+        
+
+    async def load(self) -> bool:
+        self.ready = True
+        return self.ready
+
+    async def predict(self, payload: InferenceRequest) -> InferenceResponse:
+        if self.delay:
+            time.sleep(self.delay)
+        
+        return InferenceResponse(
+            model_name=self.name,
+            model_version=self.version,
+            outputs=[
+                ResponseOutput(
+                    name=request_input.name,
+                    shape=request_input.shape,
+                    datatype=request_input.datatype,
+                    parameters=request_input.parameters,
+                    data=request_input.data
+                ) for request_input in payload.inputs
+            ]
+        )
+```
+
+The identity model only forwards the input tensors to the output and adds a delay. The delay added by the identity model is important to avoid infinite loops in our pipeline due to the joining interval used by Kafka Streams. Please check this documentation for more details.
+
+## Pipeline
+
+Our pipeline for the counter application will be composed of 3 models: the counter model, an identity feedback loop model, and an identity output model. The schema for the pipeline is the following:
+
+```mermaid
+flowchart LR
+    input([Input])
+    counter([Counter])
+    identity_loop([Identity Loop])
+    identity_output([Identity Output])
+
+    any([Any])
+    output([Output])
+
+    input --> any --> counter
+    counter --> identity_loop
+    identity_loop --> any
+    counter --> identity_output --> output
+
+    counter -. continue .-> identity_loop
+    counter -. stop .-> identity_output 
+
+    classDef any fill:#0f0,stroke:#080,stroke-width:2px,color:#000;
+    classDef main fill:#d0f0ff,stroke:#0080ff,stroke-width:2px,color:#003366;
+
+    class any any;
+    class counter,identity_loop,identity_output main;
+```
+
+## Models deployment
+
+To deploy the pipeline above, we first need to load the models into our cluster. The `model-settings.json` file for the counter model is as follows:
+
+```json
+{
+    "name": "counter",
+    "implementation": "model.Counter",
+    "parameters": {
+        "version": "v0.1.0"
+    }
+}
+```
+
+For the identity loop model, we will use the same `model-settings.json` file, but we will add a delay of 1 millisecond to the feedback loop model:
+
+```json
+{
+    "name": "identity-loop",
+    "implementation": "model.IdentityModel",
+    "parameters": {
+        "version": "v0.1.0",
+        "extra": {
+            "delay": 0.001
+        }
+    }
+}
+```
+The one-millisecond delay is important to avoid infinite loops in our pipeline and that is the joining window for any input type to our counter model, and the joining window for our identity model, which we will define in our pipeline. 
```

Comment:
```suggestion
The one-millisecond delay is crucial to prevent infinite loops in the pipeline. It aligns with the join window applied to all input types for the counter model, as well as the join window configured for the identity model, which is specified in the pipeline definition.
```